### PR TITLE
GitHub Action: Activate security analysis with CodeQL

### PR DESCRIPTION
**Pull request by neteler**
This PR adds Code-QL which is a tool for

```
Code scanning alerts
Automatically detect common vulnerability and coding errors 
```

It will run for master branch and protected branches and for MRs into the master branch. 
For more details see https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/enabling-code-scanning-for-a-repository

Developed with support by @mmacata (https://github.com/mundialis/actinia_core/pull/129)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, * ]
```
Comment (neteler) [inline .github/workflows/codeql-analysis.yml:16]:
```suggestion
    branches: [ master,  releasebranch_7_8]
```
Limit to master and releasebranch_7_8

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
```
Comment (neteler) [inline .github/workflows/codeql-analysis.yml:65]:
```suggestion
    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
    # If this step fails, then you should remove it and run the build manually (see below)
    ##- name: Autobuild
    ##  uses: github/codeql-action/autobuild@v1

    # ℹ️ Command-line programs to run using the OS shell.
    # 📚 https://git.io/JvXDl

    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
    #    and modify them (or add more) to build your code if your project
    #    uses a compiled language

    #- run: |
    #   make bootstrap
    #   make release

    - name: Get dependencies
      run: |
        sudo apt-get update -y
        sudo apt-get install -y wget git gawk findutils
        xargs -a <(awk '! /^ *(#|$)/' ".github/workflows/apt.txt") -r -- \
            sudo apt-get install -y --no-install-recommends --no-install-suggests
    - name: Create installation directory
      run: |
        mkdir $HOME/install
    - name: Ensure one core for compilation
      run: |
        echo "MAKEFLAGS=-j1" >> $GITHUB_ENV
    - name: Set LD_LIBRARY_PATH for compilation
      run: |
        echo "LD_LIBRARY_PATH=$HOME/install/lib" >> $GITHUB_ENV
    - name: Build
      env:
        # TODO: -pedantic-errors here won't go through ./configure (with GNU C)
        CFLAGS: "-std=${{ matrix.c }}"
        # TODO: -pedantic-errors here won't compile
        CXXFLAGS: "-std=${{ matrix.cpp }}"
      run: .github/workflows/build.sh $HOME/install

```

Using code snippets from https://github.com/OSGeo/grass/blob/2efac9859e99cc9981ca1fa3296b9550e984d895/.github/workflows/gcc.yml#L26 to build code

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '36 18 * * 6'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
```
Comment (neteler) [inline .github/workflows/codeql-analysis.yml:26]:
```suggestion
    runs-on: ubuntu-20.04
```
Sync to `.github/workflows/gcc.yml`

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Get dependencies
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y wget git gawk findutils
```
Comment (neteler) [inline .github/workflows/codeql-analysis.yml:69]:
```suggestion
        sudo apt-get install -y wget git gawk findutils g++ gcc make
```

**Comment by neteler** [comment]
It fails with

```
...
checking for gcc... gcc
checking whether the C compiler (gcc -std= ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.
Error: Process completed with exit code 1.
```

... no idea why.

**Comment by wenzeslaus** [comment]
What about replacing this with the GCC build and starting from there? This should at least show you what CodeQL setup step interferes with the normal build and if that's the case (AFAIU the code should be build in regular way with CodeQL).

**Comment by neteler** [comment]
Fine for me - PR changes welcome

**Comment by neteler** [comment]
I guess this should be done from scratch - closing.

**Pull request closed without merging**